src/shape.rs

```diff
@@ -1319,6 +1334,12 @@ impl ShapeLine {
         let number_of_visual_lines = visual_lines.len();
         for (index, visual_line) in visual_lines.iter().enumerate() {
             if visual_line.ranges.is_empty() {
+                layout_lines.push(LayoutLine {
+                    w: 0.0,
+                    max_ascent: 0.0,
+                    max_descent: 0.0,
+                    glyphs: Default::default(),
+                });
```

Comment:
Nit: could we add a comment explaining why we need this empty line (similar to your explanation from Slack)? Not immediately clear as a reader why this is needed - something simple should work!

Reply:
Good call, thank you! Done.